### PR TITLE
ixfrdist: add support for outgoing notify

### LINKED ISSUE #6994: Ixfrdist Notify Support
- Program: ixfrdist
 - Issue type: Feature request

### Short description
It would be nice if IXFRDist support NOTIFY, Catch DNS NOTIFY from master and update AFXR/Send DNS NOTIFY to some hosts.

### Usecase
When ixfrdist is use with auth server, this feature permit to initiate AXFR only when necessary.

### Description
ixfrdist can catch NOTIFY from master/auth, and can initiate AXFR transfer zone and SOA check only when there is modification on master server vs check SOA expiration check only.
This feature may permit to replace the limitation of PowerDNS to send IXFR.

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
Fix #6994

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/ixfrdist.cc [hunk 1]
```diff
@@ -284,6 +297,84 @@ static void updateCurrentZoneInfo(const DNSName& domain, std::shared_ptr<ixfrinf
...
+      g_log << Logger::Notice << "Communicator thread stopped" << std::endl;
+      break;
+    }
+    {
+      std::set<int> fds = {sock4, sock6};
+      ComboAddress from;
+      char buffer[1500];
+      int sock;
+
+      // receive incoming notifications on the nonblocking socket and take them off the list
```
Comment (Habbie) [inline pdns/ixfrdist.cc:330]:
```suggestion
      // receive incoming notification answers on the nonblocking sockets and take them off the list
```

?

File: pdns/communicator.hh [hunk 2]
```diff
@@ -78,7 +78,11 @@ public:
   void add(const DNSName& domain, const string& ipstring, time_t delay = 0)
   {
     const ComboAddress ipaddress(ipstring);
+    add(domain, ipaddress, delay);
+  }
 
+  void add(const DNSName& domain, const ComboAddress& ipaddress, time_t delay = 0)
```
Comment (Habbie) [inline pdns/communicator.hh:84]: nice cleanup

**Review by Habbie — COMMENTED** [review]
Early review of draft. 

Nice, I hadn't thought of reusing so much from auth. Pity this pulls in so many files, but things have always been that way.

PR seems to make sense. I'll do a full review when you're ready for that (including a test, I hope!)

**Comment by chbruyand** [comment]
Both dnsdist and recursor tests failures are unrelated to this PR.

**Review by Habbie — APPROVED** [review]

**Pull request merged**